client.go

```diff
@@ -321,18 +321,18 @@ func commitContainerToImage(containerId, repo, tag, comment, author string) erro
 // The name of the new image is specified via target_repo and target_tag.
 // The container is always deleted.
 // If something goes wrong an error message is returned.
-func runCommandAndCommitToImage(img, target_repo, target_tag, cmd, comment, author string) error {
+func runCommandAndCommitToImage(img, target_repo, target_tag, cmd, comment, author string) (string, error) {
 	containerId, err := runCommandInContainer(img, []string{cmd}, true)
-	if err != nil {
-		return err
+	if err != nil || containerId == ""{
```

Comment:
If `containerId == ""`, it's guaranteed that err will not be nil. Since the error should be descriptive enough, I'd say to keep it as it was.